xen/arch/arm/p2m.c

```diff
@@ -1098,6 +1102,34 @@ int p2m_set_entry(struct p2m_domain *p2m,
         nr -= (1 << order);
     }
 
+    if ( likely(!rc) )
```

Comment:
as __p2m_set_entry is called in the loop above and SOME of the p2m entries may be set. With this check we will only update IOMMU when ALL p2m entries are set. This will make p2m not synchronized CPU<>IOMMU
Consider moving IOMMU into while loop and using map_page, not map_pages? Or track number of pages to be used by IOMMU